Sources/Workspace/WorkspaceConfiguration.swift

```diff
@@ -209,6 +209,107 @@ extension Workspace {
     }
 }
 
+// MARK: - Authorization
+
+extension Workspace {
+    public struct Authorization {
+        private let configuration: Configuration
+
+        public init(configuration: Configuration) {
+            self.configuration = configuration
+        }
+
+        public func makeAuthorizationProvider(fileSystem: FileSystem, observabilityScope: ObservabilityScope) throws -> AuthorizationProvider? {
+            var providers = [AuthorizationProvider]()
+
+            switch self.configuration.netrc {
+            case .custom(let path):
+                guard fileSystem.exists(path) else {
+                    throw StringError("Did not find .netrc file at \(path).")
+                }
+                providers.append(try NetrcAuthorizationProvider(path: path, fileSystem: localFileSystem))
+            case .`default`(let rootPath):
+                // package/project "local" .netrc file, takes priority over user-level file
+                let localPath = rootPath.appending(component: ".netrc")
+                // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
+                if let localProvider = self.tryLoadNetrc(path: localPath, fileSystem: fileSystem, observabilityScope: observabilityScope) {
+                    providers.append(localProvider)
+                }
+
+                // user .netrc file (most typical)
+                let userHomePath = fileSystem.homeDirectory.appending(component: ".netrc")
```

Comment:
Please wait for the outcome of https://github.com/apple/swift-package-manager/pull/4075 before merging this